tensorflow/lite/micro/micro_resource_variable.cc

```diff
@@ -92,6 +92,9 @@ TfLiteStatus MicroResourceVariables::Allocate(int id, TfLiteContext* context,
     variable.bytes = tensor->bytes;
     variable.resource_buffer =
         context->AllocatePersistentBuffer(context, tensor->bytes);
+    // Zero out resource buffers by deafult. Buffers can be initialized to
+    // nonzero values using ASSIGN_VARIABLE.
+    memset(variable.resource_buffer, 0, variable.bytes);
```

Comment:
Does it make more sense if we check whether the resource buffer is allocated successfully before calling memset() on it?

Reply:
Good call. Move to after the null check.